internal/byok/command_create.go

```diff
@@ -43,15 +41,62 @@ var encryptionKeyPolicyAws = template.Must(template.New("encryptionKeyPolicyAws"
 }`))
 
 const (
+	failedToRenderKeyPolicyErrorMsg = "BYOK error: failed to render key policy"
+
 	keyVaultCryptoServiceEncryptionUser = "e147488a-f6f5-4113-8e2d-b22465e65bf6"
 	keyVaultReader                      = "21090545-7ca7-4776-b22c-e363652d74d2"
+
+	defaultGcpRoleName = "custom_kms_role"
 )
 
+type gcpPolicyMetadata struct {
+	project  string
+	location string
+	keyRing  string
+	key      string
+	group    string
+}
+
+func (g gcpPolicyMetadata) renderPolicy() string {
+	return fmt.Sprintf(`
+gcloud iam roles create %[1]s \
+	--project=%[2]s \
+	--description="Grant necessary permissions for Confluent to access KMS key" \
+	--permissions=cloudkms.cryptoKeyVersions.useToDecrypt,cloudkms.cryptoKeyVersions.useToEncrypt,cloudkms.cryptoKeys.get  && \
+gcloud kms keys add-iam-policy-binding %[3]s \
+	--project=%[2]s \
+	--keyring='%[4]s' \
+	--location='%[5]s' \
+	--member='group:%[6]s' \
+	--role='projects/%[2]s/roles/%[1]s'`, g.getCustomRoleName(), g.project, g.key, g.keyRing, g.location, g.group)
+}
+
+func newGcpPolicyMetadata(project, location, keyRing, key, group string) gcpPolicyMetadata {
+	return gcpPolicyMetadata{
+		project:  project,
+		location: location,
+		keyRing:  keyRing,
+		key:      key,
+		group:    group,
+	}
+}
```

Comment:
Turning this into a function just makes this less readable since you can't tell what all the field names are, IMO